### PR TITLE
Issue #19: Have pftt delete the files created to run the test if the test pass.

### PR DESCRIPTION
**Pull request by lavturo**
For the most part, I was trying to reflect how run-test.php runs with these changes.

### CODE REVIEW

**Comment by lavturo** [comment]
@weltling @hollyhuiLi Can you please check this? 

Some notes:
- For PhptTestPreparer, I removed the reference to AzureWebsitesScenario because there were many errors in the file. Took out the only section that was used for creating the cleaning file. 

- For AbstractPhptTestCaseRunner, it was checking if the host was busy to determine if it should run the cleaning file. The way it checked was by seeing if an AtomicInteger was less than 192 (If so, then host is busy). When I debugged it, I never saw the AtomicInteger go above 1, so it never ran the cleaning file. I removed it for the time being and it seems to rune fine. 

- One thing I didn't check was to see if it took longer than before to run the tests. It ran fine, but I didn't time it. I believe it should be slightly slower because it's trying to clean files as well, but it should still be faster than running tests manually.

**Comment by weltling** [comment]
@lavturo thanks for the PR. Not taking in account the Azure site scenario should be ok, for now. 

The `isBusy()` method looks a bit strange. Say, if tests are running, then logically it would be `num_tests > 0` - busy. What `num tests < 192` could mean? We should clarify, why the condition is done this way and under which circumstances would the counter go beyond as in your case.

Also - the first change b423a5d0587eb4f97f67f2fc1434d4b99f37ba05 was about cleaning out the files at startup. This patch would execute cleanup after every test run. Wouldn't this remove files for other tests running in parallel? Which number of threads do you usually use?

Thanks.

**Comment by lavturo** [comment]
@weltling I don't believe this would remove files for other tests that are running in parallel. Some test files have a CLEAN section in the test and this change is to run that CLEAN section. It should only be removing files and directories that are related to the test file if the run succeeds. I usually use the default amount (16), or 8 threads. 

As for the isBusy() method, as you have stated, logically it would make sense like that (where it should be busy if num_tests > 0). I can try to look into it more soon, but there isn't much to go off of.

File: src/com/mostc/pftt/model/core/PhptSourceTestPack.java [hunk 1]
```diff
@@ -142,6 +142,7 @@ public void cleanup(ConsoleManager cm) {
 		fs.deleteFileExtension(test_pack, ".skip.php");
 		fs.deleteFileExtension(test_pack, ".cmd");
 		fs.deleteFileExtension(test_pack, ".sh");
+		fs.deleteFileExtension(test_pack, ".clean.php");
 		fs.deleteFileExtension(test_pack, ".tmp");
```
Comment (weltling) [inline src/com/mostc/pftt/model/core/PhptSourceTestPack.java]: @lavturo but this change is not related to the `--CLEAN--` section, right? What you indeed touch is how `executeClean()` is called. That confused me a bit :)

Thanks.

Comment (lavturo) [reply src/com/mostc/pftt/model/core/PhptSourceTestPack.java]: There is also a change I made where it creates the ".clean.php" (`PhptTestPreparer.java`), which is why this change is here. In case the test fails and it doesn't get deleted, this cleanup is supposed to delete the leftover files. 

`executeClean()` is located in `CliPhptTestCaseRunner.java `where it references the location of the ".clean.php" file to run. It currently works as is. 

Hope that clears the confusion.

**Comment by weltling** [comment]
Merged, thanks!
